File: yt/yt/flow/examples/java/wait_click_join/wait_click_join/src/main/java/tech/ytsaurus/flow/examples/waitclickjoin/model/JoinedAction.java
package tech.ytsaurus.flow.examples.waitclickjoin.model;

import javax.persistence.Column;
import javax.persistence.Entity;

// [BEGIN joined_action_model]
// The stream schema is derived from this class: the runner writes it into the pipeline spec,
// so "joined_action" needs no schema of its own there. Field order defines column order, and
// columnDefinition pins a column type where the default for the field type differs.
@Entity
public class JoinedAction {
    @Column(name = "hit_id", columnDefinition = "string")
    private String hitId;
    @Column(name = "hit_time", columnDefinition = "uint64")
    private Long hitTime;
    @Column(name = "is_click")
    private Boolean isClick;
    @Column(name = "show_time", columnDefinition = "uint64")
    private Long showTime;
    @Column(name = "click_time", columnDefinition = "uint64")
    private Long clickTime;
    @Column(name = "hit_payload", columnDefinition = "string")
    private String hitPayload;

    public JoinedAction() {
    }

    public JoinedAction(String hitId, Long hitTime, Boolean isClick, Long showTime, Long clickTime, String hitPayload) {
        this.hitId = hitId;
        this.hitTime = hitTime;
        this.isClick = isClick;
        this.showTime = showTime;
        this.clickTime = clickTime;
        this.hitPayload = hitPayload;
    }

    public String getHitId() {
        return hitId;
    }

    public void setHitId(String hitId) {
        this.hitId = hitId;
    }

    public Long getHitTime() {
        return hitTime;
    }

    public void setHitTime(Long hitTime) {
        this.hitTime = hitTime;
    }

    public Boolean getClick() {
        return isClick;
    }

    public void setClick(Boolean click) {
        isClick = click;
    }

    public Long getShowTime() {
        return showTime;
    }

    public void setShowTime(Long showTime) {
        this.showTime = showTime;
    }

    public Long getClickTime() {
        return clickTime;
    }

    public void setClickTime(Long clickTime) {
        this.clickTime = clickTime;
    }

    public String getHitPayload() {
        return hitPayload;
    }

    public void setHitPayload(String hitPayload) {
        this.hitPayload = hitPayload;
    }
}
// [END joined_action_model]

File: yt/yt/flow/examples/kotlin/wait_click_join/wait_click_join/src/main/kotlin/tech/ytsaurus/flow/examples/waitclickjoin/model/JoinedAction.kt
package tech.ytsaurus.flow.examples.waitclickjoin.model

import javax.persistence.Column
import javax.persistence.Entity

// [BEGIN joined_action_model]
// The stream schema is derived from this class: the runner writes it into the pipeline spec,
// so "joined_action" needs no schema of its own there. Field order defines column order, and
// columnDefinition pins a column type where the default for the field type differs.
@Entity
class JoinedAction {
    @Column(name = "hit_id", columnDefinition = "string")
    var hitId: String? = null

    @Column(name = "hit_time", columnDefinition = "uint64")
    var hitTime: Long? = null

    @get:JvmName("getClick")
    @set:JvmName("setClick")
    @Column(name = "is_click")
    var isClick: Boolean? = null

    @Column(name = "show_time", columnDefinition = "uint64")
    var showTime: Long? = null

    @Column(name = "click_time", columnDefinition = "uint64")
    var clickTime: Long? = null

    @Column(name = "hit_payload", columnDefinition = "string")
    var hitPayload: String? = null

    constructor()

    constructor(hitId: String, hitTime: Long, isClick: Boolean, showTime: Long, clickTime: Long, hitPayload: String) {
        this.hitId = hitId
        this.hitTime = hitTime
        this.isClick = isClick
        this.showTime = showTime
        this.clickTime = clickTime
        this.hitPayload = hitPayload
    }
}
// [END joined_action_model]
